### ISSUE TITLE
<audio> inside <p> causes an empty <p>

### ISSUE THREAD

**Issue by Crusse** [Masterminds/html5-php#121]
I'm using version 2.2.2 with PHP PHP 5.4.45.

This code

```
$html = <<<HTML
<!DOCTYPE html>

  <p><audio preload="none" controls="controls">
    <source src="https://example.com/test.mp3" type="audio/mpeg" />
    Your browser does not support the audio element.
  </audio></p>
HTML;

$parser = new Masterminds\HTML5( array( 'disable_html_ns' => true ) );
$doc = $parser->loadHTML( $html );
echo $parser->saveHTML( $doc );
```

results in this HTML

```
<!DOCTYPE html>
<html><p></p><audio preload="none" controls="controls">
    <source src="https://example.com/test.mp3" type="audio/mpeg">
    Your browser does not support the audio element.
  </audio></html>
```

Running this in Chrome says that `<audio>` is an inline element (at least in this context):

```
<!DOCTYPE html>

  <p><audio preload="none" controls="controls">
    <source src="https://example.com/test.mp3" type="audio/mpeg" />
    Your browser does not support the audio element.
  </audio></p>


<script>
setTimeout( function() {
  var c = document.getElementsByTagName('audio')[0];
  alert(getComputedStyle(c, null).display);
}, 1000 );
</script>
```

**Comment by goetas** [comment]
I'm not sure that the tag p can contain  audio tags... is it true?

**Comment by Crusse** [comment]
I'm not sure either. My use case is with TinyMCE, which actually adds p tags around audio tags: https://jsfiddle.net/z35qy9h0/1/

A random SO thread: http://stackoverflow.com/questions/9852312/list-of-html5-elements-that-can-be-nested-inside-p-element

Also testing the aforementioned HTML in Firefox and Chrome results in the audio element staying inside the p element, when looking at the DOM, whereas if there was a div inside a p element, those browsers would render it outside of the p element.

**Comment by Lewiscowles1986** [comment]
Is turning off autop not a better option than adding edge cases to other software?

**Comment by mattfarina** [comment]
This one is a little more complicated once I started looking at the specs...

In [HTML 4.01 is notes](https://www.w3.org/TR/html401/struct/text.html#h-9.3.1),
> The P element represents a paragraph. It cannot contain block-level elements (including P itself).

This exact line doesn't show up in the HTML5 spec.

The HTML5 spec does say,
> Runs of phrasing content form paragraphs.

[Phrasing content](https://www.w3.org/TR/html5/dom.html#phrasing-content-1) does contain audio tags.

So, this is a bug.

While I've yet to track it completely, I did find that it's in the parser. The next step is to start looking at how p auto-closing works.
